{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "51252a02-bb76-485a-a5f0-203038b36172",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Importing the required libraries\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "from sklearn.model_selection import train_test_split, RandomizedSearchCV\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "from sklearn.metrics import accuracy_score, balanced_accuracy_score, classification_report, precision_recall_curve\n",
    "from sklearn.metrics import PrecisionRecallDisplay, confusion_matrix, ConfusionMatrixDisplay, mean_squared_error, r2_score\n",
    "from sklearn.metrics import mean_absolute_error, median_absolute_error, explained_variance_score\n",
    "from sklearn.metrics import precision_score, recall_score, f1_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b90802bc-ea19-49dc-9fab-861542abb82b",
   "metadata": {},
   "outputs": [],
   "source": [
    "Mutational Signature Dataset - Tally file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "19e9b2e9-4207-4b20-93de-e29937df3f8b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "--------------------------\n",
      "\n",
      "channel           sample   A[C>A]A   A[C>A]C   A[C>A]G  A[C>A]T   A[C>G]A  \\\n",
      "0        TCGA-A2-A0T5-01  0.000000  0.000000  0.000000  0.00000  0.000000   \n",
      "1        TCGA-CF-A9FF-01  0.000000  0.010811  0.010811  0.00000  0.000000   \n",
      "2        TCGA-CA-6717-01  0.000848  0.001774  0.000309  0.01049  0.000848   \n",
      "\n",
      "channel   A[C>G]C   A[C>G]G  A[C>G]T   A[C>T]A  ...   T[T>A]T   T[T>C]A  \\\n",
      "0        0.000000  0.000000  0.00000  0.000000  ...  0.000000  0.000000   \n",
      "1        0.000000  0.010811  0.00000  0.010811  ...  0.000000  0.005405   \n",
      "2        0.000848  0.000000  0.00054  0.001465  ...  0.002854  0.006788   \n",
      "\n",
      "channel   T[T>C]C   T[T>C]G   T[T>C]T   T[T>G]A   T[T>G]C  T[T>G]G   T[T>G]T  \\\n",
      "0        0.000000  0.000000  0.000000  0.000000  0.000000   0.0000  0.000000   \n",
      "1        0.000000  0.000000  0.000000  0.000000  0.000000   0.0000  0.005405   \n",
      "2        0.017123  0.009873  0.017663  0.007636  0.009873   0.0027  0.082298   \n",
      "\n",
      "channel  p53 status  \n",
      "0                    \n",
      "1                    \n",
      "2                    \n",
      "\n",
      "[3 rows x 98 columns]\n",
      "\n",
      "--------------------------\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#############################\n",
    "#### Signature Database #####\n",
    "#############################\n",
    "folder_path = '../../../Downloads/new_sigs/'\n",
    "sig_data = pd.DataFrame()\n",
    "\n",
    "# Loop through all CSV files in the folder\n",
    "for filename in os.listdir(folder_path):\n",
    "    # Filter to match the pattern 'SBS96_catalogue.<sample_name>.hg19.tally.csv' -  only doing sbs first\n",
    "    if filename.startswith('SBS96_catalogue.') and filename.endswith('.hg19.tally.csv'):\n",
    "        file_path = os.path.join(folder_path, filename)\n",
    "        data = pd.read_csv(file_path, index_col=None)\n",
    "\n",
    "        # Extract the sample name from the filename (e.g., 'TCGA-CA-6717-01')\n",
    "        sample_name = filename.split('.')[1]\n",
    "        select_data = data.drop(columns=['type', 'count'])\n",
    "\n",
    "        # Pivot the DataFrame so that 'channel' becomes the columns\n",
    "        pivot_data = select_data.pivot_table(index=None, columns=\"channel\", values=\"fraction\").reset_index(drop=True)\n",
    "        pivot_data.insert(0, 'sample', sample_name)\n",
    "        sig_data = pd.concat([sig_data, pivot_data], ignore_index=True)\n",
    "\n",
    "# Add final column to data (empty) - maybe dont do it here??\n",
    "sig_data[\"p53 status\"] = \"\n",
    "\n",
    "# Display the final DataFrame or save it to a file\n",
    "print(\"\\n--------------------------\\n\")\n",
    "print(sig_data)\n",
    "print(\"\\n--------------------------\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "482b7630-7898-4f6e-90cf-2230a084cd7d",
   "metadata": {},
   "source": [
    "MAF File: https://xenabrowser.net/datapages/?dataset=October_2016_whitelist_2583.snv_mnv_indel.maf.xena.nonUS&host=https%3A%2F%2Fpcawg.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "0279964d-e140-4229-8fd3-c77f9ced0e1e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/pr/sxpxd22s7jj41b9d9c6lk5yc0000gp/T/ipykernel_34917/2570192616.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  filtered_data['HG19_Variant'] = 'chr17:g.' + \\\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           Sample          HG19_Variant      gene             effect\n",
      "0         DO46416    chr17:g.1230447G>A     ACAP3             Intron\n",
      "1         DO46416    chr17:g.1609722C>T  SLC35E2B             Intron\n",
      "2         DO46416    chr17:g.1903275C>T   Unknown                IGR\n",
      "3         DO46416    chr17:g.2574998C>T     TTC34             Intron\n",
      "4         DO46416    chr17:g.3151249G>A    PRDM16             Intron\n",
      "...           ...                   ...       ...                ...\n",
      "21980512  DO51503  chr17:g.150611902G>A   Unknown                IGR\n",
      "21980513  DO51503  chr17:g.151231465T>C   Unknown                IGR\n",
      "21980514  DO51503  chr17:g.151815421C>T     GABRQ  Missense_Mutation\n",
      "21980515  DO51503  chr17:g.152000524T>C     NSDHL             Intron\n",
      "21980516  DO51503  chr17:g.152538786G>A   Unknown                IGR\n",
      "\n",
      "[20621575 rows x 4 columns]\n"
     ]
    }
   ],
   "source": [
    "#############################\n",
    "######### MAF FILE ##########\n",
    "#############################\n",
    "maf_file_path = '../../../Downloads/October_2016_whitelist_2583.snv_mnv_indel.maf.xena.nonUS'\n",
    "maf_data = pd.read_csv(maf_file_path, sep='\\t', comment='#', low_memory=False)\n",
    "\n",
    "filtered_data = maf_data[(maf_data['start'] == maf_data['end']) & (maf_data['alt'] != '-')]\n",
    "filtered_data['HG19_Variant'] = 'chr17:g.' + \\\n",
    "                                (filtered_data['start'] - 1).astype(str) + \\\n",
    "                                filtered_data['reference'] + '>' + \\\n",
    "                                filtered_data['alt']\n",
    "\n",
    "maf_filtered = filtered_data[['Sample', 'HG19_Variant', 'gene', 'effect']]\n",
    "print(maf_filtered)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "180e1bc0-7f87-40ad-aafc-9cc49493eedf",
   "metadata": {},
   "source": [
    "Dataset: Download at https://p53.fr/download-the-database - Using .xlsx EU Variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "6f97c1bc-ad8c-44ea-961c-ee52a5283c09",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/asanghvi/Desktop/ML/ML_practice/.venv/lib/python3.12/site-packages/openpyxl/worksheet/_reader.py:329: UserWarning: Unknown extension is not supported and will be removed\n",
      "  warn(msg)\n",
      "/var/folders/pr/sxpxd22s7jj41b9d9c6lk5yc0000gp/T/ipykernel_34917/465436135.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  tester_data_filtered['type'] = tester_data_filtered['Mutational_event']\n"
     ]
    }
   ],
   "source": [
    "#############################\n",
    "####### TP53 Database #######\n",
    "#############################\n",
    "new_file_tester =  '../../../Downloads/UMD_variants_EU.xlsx'\n",
    "tester_data =  pd.read_excel(new_file_tester)\n",
    "\n",
    "pattern = r'^chr17:g\\.\\d+[A, C, G, T]>[A, C, G, T]$'\n",
    "tester_data_filtered = tester_data[tester_data['HG19_Variant'].str.contains(pattern, regex=True)]\n",
    "tester_data_filtered['type'] = tester_data_filtered['Mutational_event']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "id": "a7de7f5b-8cf3-4317-9bfe-573a52c2be80",
   "metadata": {},
   "outputs": [],
   "source": [
    "p53_db = tester_data_filtered[['HG19_Variant', 'Pathogenicity', 'Final comment']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ce17008-13c1-4458-b4f7-bd22c44ea2e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#############################\n",
    "###### Join Dataframes ######\n",
    "#############################\n",
    "merged_data = pd.merge(maf_filtered, p53_db[['HG19_Variant', 'Pathogenicity', 'Final comment']], \n",
    "                       on='HG19_Variant', how='left')\n",
    "merged_data.fillna({ 'Pathogenicity' : 'Unknown' }, inplace=True)\n",
    "merged_data.fillna({ 'Final comment' : 'No Comment' }, inplace=True)\n",
    "\n",
    "pathogenicity_mapping = {\n",
    "    'Pathogenic': 1,\n",
    "    'Likely Pathogenic': 0.75,\n",
    "    'Possibly pathogenic': 0.5,\n",
    "    'VUS': 0.25,\n",
    "    'Benign': 0,\n",
    "    'Unknown': -1,\n",
    "}\n",
    "\n",
    "merged_data['p53 status'] = merged_data['Pathogenicity'].map(pathogenicity_mapping)\n",
    "formatted_merged = merged_data[['Sample', 'HG19_Variant', 'gene', 'Pathogenicity', 'Final comment', 'p53 status']]\n",
    "prepared_data = formatted_merged[(formatted_merged['p53 status'] != -1.0) & (formatted_merged['gene'] == 'TP53')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "bb5085ab-1711-4d75-b0e4-041fb08268b4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           Sample          HG19_Variant      gene             effect  \\\n",
      "0         DO46416    chr17:g.1230447G>A     ACAP3             Intron   \n",
      "1         DO46416    chr17:g.1609722C>T  SLC35E2B             Intron   \n",
      "2         DO46416    chr17:g.1903275C>T   Unknown                IGR   \n",
      "3         DO46416    chr17:g.2574998C>T     TTC34             Intron   \n",
      "4         DO46416    chr17:g.3151249G>A    PRDM16             Intron   \n",
      "...           ...                   ...       ...                ...   \n",
      "20621570  DO51503  chr17:g.150611902G>A   Unknown                IGR   \n",
      "20621571  DO51503  chr17:g.151231465T>C   Unknown                IGR   \n",
      "20621572  DO51503  chr17:g.151815421C>T     GABRQ  Missense_Mutation   \n",
      "20621573  DO51503  chr17:g.152000524T>C     NSDHL             Intron   \n",
      "20621574  DO51503  chr17:g.152538786G>A   Unknown                IGR   \n",
      "\n",
      "         Pathogenicity Final comment  p53 status  \n",
      "0              Unknown    No Comment        -1.0  \n",
      "1              Unknown    No Comment        -1.0  \n",
      "2              Unknown    No Comment        -1.0  \n",
      "3              Unknown    No Comment        -1.0  \n",
      "4              Unknown    No Comment        -1.0  \n",
      "...                ...           ...         ...  \n",
      "20621570       Unknown    No Comment        -1.0  \n",
      "20621571       Unknown    No Comment        -1.0  \n",
      "20621572       Unknown    No Comment        -1.0  \n",
      "20621573       Unknown    No Comment        -1.0  \n",
      "20621574       Unknown    No Comment        -1.0  \n",
      "\n",
      "[20621575 rows x 7 columns]\n"
     ]
    }
   ],
   "source": [
    "print(merged_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "8daf0a40-a276-40b7-a28a-4dd7d8ad586f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      Sample Pathogenicity\n",
      "0     DO1000        Benign\n",
      "1     DO1001    Pathogenic\n",
      "2     DO1002        Benign\n",
      "3     DO1003        Benign\n",
      "4     DO1004        Benign\n",
      "...      ...           ...\n",
      "1777  DO7196        Benign\n",
      "1778  DO7214        Benign\n",
      "1779  DO7280        Benign\n",
      "1780  DO7304        Benign\n",
      "1781  DO7328        Benign\n",
      "\n",
      "[1782 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/pr/sxpxd22s7jj41b9d9c6lk5yc0000gp/T/ipykernel_34917/4119386144.py:3: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  pathogenicity_status = df.groupby('Sample').apply(\n"
     ]
    }
   ],
   "source": [
    "def assign_pathogenicity_status(df):\n",
    "    # Group by 'Sample' and determine pathogenicity for each group\n",
    "    pathogenicity_status = df.groupby('Sample').apply(\n",
    "        lambda group: 'Pathogenic' if ((group['gene'] == 'TP53') & \n",
    "                                       (group['Pathogenicity'].isin(['Pathogenic', 'Likely Pathogenic']))).any() \n",
    "                      else 'Benign'\n",
    "    ).reset_index(name='Pathogenicity')\n",
    "    \n",
    "    return pathogenicity_status\n",
    "\n",
    "new_merged_data = assign_pathogenicity_status(merged_data)\n",
    "\n",
    "print(new_merged_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "2b218b24-ffc9-4ade-834e-f979ad945f1b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      Sample  p53 status\n",
      "0     DO1000           0\n",
      "1     DO1001           1\n",
      "2     DO1002           0\n",
      "3     DO1003           0\n",
      "4     DO1004           0\n",
      "...      ...         ...\n",
      "1777  DO7196           0\n",
      "1778  DO7214           0\n",
      "1779  DO7280           0\n",
      "1780  DO7304           0\n",
      "1781  DO7328           0\n",
      "\n",
      "[1782 rows x 2 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/pr/sxpxd22s7jj41b9d9c6lk5yc0000gp/T/ipykernel_34917/2780195078.py:3: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  pathogenicity_status = df.groupby('Sample').apply(\n"
     ]
    }
   ],
   "source": [
    "def assign_pathogenicity_status(df):\n",
    "    # Group by 'Sample' and assign p53 status as a numeric value (1 for pathogenic, 0 for benign)\n",
    "    pathogenicity_status = df.groupby('Sample').apply(\n",
    "        lambda group: 1 if ((group['gene'] == 'TP53') & \n",
    "                            (group['p53 status'].astype(float) > 0.5)).any() \n",
    "                      else 0\n",
    "    ).reset_index(name='p53 status')\n",
    "    \n",
    "    return pathogenicity_status\n",
    "\n",
    "# Run the function and print the result\n",
    "new_merged_data_temp = assign_pathogenicity_status(merged_data)\n",
    "print(new_merged_data_temp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "00ec89c5-cf7d-4bfa-93bd-5ea0cb45e277",
   "metadata": {},
   "outputs": [],
   "source": [
    "##############################\n",
    "###### Machine Learning ######\n",
    "##############################\n",
    "X = new_merged_data_temp.drop(columns=['Sample'])\n",
    "y = new_merged_data_temp['p53 status']\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "55428ebc-ef0e-41b1-a046-7735538c03f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>#sk-container-id-2 {\n",
       "  /* Definition of color scheme common for light and dark mode */\n",
       "  --sklearn-color-text: black;\n",
       "  --sklearn-color-line: gray;\n",
       "  /* Definition of color scheme for unfitted estimators */\n",
       "  --sklearn-color-unfitted-level-0: #fff5e6;\n",
       "  --sklearn-color-unfitted-level-1: #f6e4d2;\n",
       "  --sklearn-color-unfitted-level-2: #ffe0b3;\n",
       "  --sklearn-color-unfitted-level-3: chocolate;\n",
       "  /* Definition of color scheme for fitted estimators */\n",
       "  --sklearn-color-fitted-level-0: #f0f8ff;\n",
       "  --sklearn-color-fitted-level-1: #d4ebff;\n",
       "  --sklearn-color-fitted-level-2: #b3dbfd;\n",
       "  --sklearn-color-fitted-level-3: cornflowerblue;\n",
       "\n",
       "  /* Specific color for light theme */\n",
       "  --sklearn-color-text-on-default-background: var(--sg-text-color, var(--theme-code-foreground, var(--jp-content-font-color1, black)));\n",
       "  --sklearn-color-background: var(--sg-background-color, var(--theme-background, var(--jp-layout-color0, white)));\n",
       "  --sklearn-color-border-box: var(--sg-text-color, var(--theme-code-foreground, var(--jp-content-font-color1, black)));\n",
       "  --sklearn-color-icon: #696969;\n",
       "\n",
       "  @media (prefers-color-scheme: dark) {\n",
       "    /* Redefinition of color scheme for dark theme */\n",
       "    --sklearn-color-text-on-default-background: var(--sg-text-color, var(--theme-code-foreground, var(--jp-content-font-color1, white)));\n",
       "    --sklearn-color-background: var(--sg-background-color, var(--theme-background, var(--jp-layout-color0, #111)));\n",
       "    --sklearn-color-border-box: var(--sg-text-color, var(--theme-code-foreground, var(--jp-content-font-color1, white)));\n",
       "    --sklearn-color-icon: #878787;\n",
       "  }\n",
       "}\n",
       "\n",
       "#sk-container-id-2 {\n",
       "  color: var(--sklearn-color-text);\n",
       "}\n",
       "\n",
       "#sk-container-id-2 pre {\n",
       "  padding: 0;\n",
       "}\n",
       "\n",
       "#sk-container-id-2 input.sk-hidden--visually {\n",
       "  border: 0;\n",
       "  clip: rect(1px 1px 1px 1px);\n",
       "  clip: rect(1px, 1px, 1px, 1px);\n",
       "  height: 1px;\n",
       "  margin: -1px;\n",
       "  overflow: hidden;\n",
       "  padding: 0;\n",
       "  position: absolute;\n",
       "  width: 1px;\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-dashed-wrapped {\n",
       "  border: 1px dashed var(--sklearn-color-line);\n",
       "  margin: 0 0.4em 0.5em 0.4em;\n",
       "  box-sizing: border-box;\n",
       "  padding-bottom: 0.4em;\n",
       "  background-color: var(--sklearn-color-background);\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-container {\n",
       "  /* jupyter's `normalize.less` sets `[hidden] { display: none; }`\n",
       "     but bootstrap.min.css set `[hidden] { display: none !important; }`\n",
       "     so we also need the `!important` here to be able to override the\n",
       "     default hidden behavior on the sphinx rendered scikit-learn.org.\n",
       "     See: https://github.com/scikit-learn/scikit-learn/issues/21755 */\n",
       "  display: inline-block !important;\n",
       "  position: relative;\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-text-repr-fallback {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       "div.sk-parallel-item,\n",
       "div.sk-serial,\n",
       "div.sk-item {\n",
       "  /* draw centered vertical line to link estimators */\n",
       "  background-image: linear-gradient(var(--sklearn-color-text-on-default-background), var(--sklearn-color-text-on-default-background));\n",
       "  background-size: 2px 100%;\n",
       "  background-repeat: no-repeat;\n",
       "  background-position: center center;\n",
       "}\n",
       "\n",
       "/* Parallel-specific style estimator block */\n",
       "\n",
       "#sk-container-id-2 div.sk-parallel-item::after {\n",
       "  content: \"\";\n",
       "  width: 100%;\n",
       "  border-bottom: 2px solid var(--sklearn-color-text-on-default-background);\n",
       "  flex-grow: 1;\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-parallel {\n",
       "  display: flex;\n",
       "  align-items: stretch;\n",
       "  justify-content: center;\n",
       "  background-color: var(--sklearn-color-background);\n",
       "  position: relative;\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-parallel-item {\n",
       "  display: flex;\n",
       "  flex-direction: column;\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-parallel-item:first-child::after {\n",
       "  align-self: flex-end;\n",
       "  width: 50%;\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-parallel-item:last-child::after {\n",
       "  align-self: flex-start;\n",
       "  width: 50%;\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-parallel-item:only-child::after {\n",
       "  width: 0;\n",
       "}\n",
       "\n",
       "/* Serial-specific style estimator block */\n",
       "\n",
       "#sk-container-id-2 div.sk-serial {\n",
       "  display: flex;\n",
       "  flex-direction: column;\n",
       "  align-items: center;\n",
       "  background-color: var(--sklearn-color-background);\n",
       "  padding-right: 1em;\n",
       "  padding-left: 1em;\n",
       "}\n",
       "\n",
       "\n",
       "/* Toggleable style: style used for estimator/Pipeline/ColumnTransformer box that is\n",
       "clickable and can be expanded/collapsed.\n",
       "- Pipeline and ColumnTransformer use this feature and define the default style\n",
       "- Estimators will overwrite some part of the style using the `sk-estimator` class\n",
       "*/\n",
       "\n",
       "/* Pipeline and ColumnTransformer style (default) */\n",
       "\n",
       "#sk-container-id-2 div.sk-toggleable {\n",
       "  /* Default theme specific background. It is overwritten whether we have a\n",
       "  specific estimator or a Pipeline/ColumnTransformer */\n",
       "  background-color: var(--sklearn-color-background);\n",
       "}\n",
       "\n",
       "/* Toggleable label */\n",
       "#sk-container-id-2 label.sk-toggleable__label {\n",
       "  cursor: pointer;\n",
       "  display: block;\n",
       "  width: 100%;\n",
       "  margin-bottom: 0;\n",
       "  padding: 0.5em;\n",
       "  box-sizing: border-box;\n",
       "  text-align: center;\n",
       "}\n",
       "\n",
       "#sk-container-id-2 label.sk-toggleable__label-arrow:before {\n",
       "  /* Arrow on the left of the label */\n",
       "  content: \"▸\";\n",
       "  float: left;\n",
       "  margin-right: 0.25em;\n",
       "  color: var(--sklearn-color-icon);\n",
       "}\n",
       "\n",
       "#sk-container-id-2 label.sk-toggleable__label-arrow:hover:before {\n",
       "  color: var(--sklearn-color-text);\n",
       "}\n",
       "\n",
       "/* Toggleable content - dropdown */\n",
       "\n",
       "#sk-container-id-2 div.sk-toggleable__content {\n",
       "  max-height: 0;\n",
       "  max-width: 0;\n",
       "  overflow: hidden;\n",
       "  text-align: left;\n",
       "  /* unfitted */\n",
       "  background-color: var(--sklearn-color-unfitted-level-0);\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-toggleable__content.fitted {\n",
       "  /* fitted */\n",
       "  background-color: var(--sklearn-color-fitted-level-0);\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-toggleable__content pre {\n",
       "  margin: 0.2em;\n",
       "  border-radius: 0.25em;\n",
       "  color: var(--sklearn-color-text);\n",
       "  /* unfitted */\n",
       "  background-color: var(--sklearn-color-unfitted-level-0);\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-toggleable__content.fitted pre {\n",
       "  /* unfitted */\n",
       "  background-color: var(--sklearn-color-fitted-level-0);\n",
       "}\n",
       "\n",
       "#sk-container-id-2 input.sk-toggleable__control:checked~div.sk-toggleable__content {\n",
       "  /* Expand drop-down */\n",
       "  max-height: 200px;\n",
       "  max-width: 100%;\n",
       "  overflow: auto;\n",
       "}\n",
       "\n",
       "#sk-container-id-2 input.sk-toggleable__control:checked~label.sk-toggleable__label-arrow:before {\n",
       "  content: \"▾\";\n",
       "}\n",
       "\n",
       "/* Pipeline/ColumnTransformer-specific style */\n",
       "\n",
       "#sk-container-id-2 div.sk-label input.sk-toggleable__control:checked~label.sk-toggleable__label {\n",
       "  color: var(--sklearn-color-text);\n",
       "  background-color: var(--sklearn-color-unfitted-level-2);\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-label.fitted input.sk-toggleable__control:checked~label.sk-toggleable__label {\n",
       "  background-color: var(--sklearn-color-fitted-level-2);\n",
       "}\n",
       "\n",
       "/* Estimator-specific style */\n",
       "\n",
       "/* Colorize estimator box */\n",
       "#sk-container-id-2 div.sk-estimator input.sk-toggleable__control:checked~label.sk-toggleable__label {\n",
       "  /* unfitted */\n",
       "  background-color: var(--sklearn-color-unfitted-level-2);\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-estimator.fitted input.sk-toggleable__control:checked~label.sk-toggleable__label {\n",
       "  /* fitted */\n",
       "  background-color: var(--sklearn-color-fitted-level-2);\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-label label.sk-toggleable__label,\n",
       "#sk-container-id-2 div.sk-label label {\n",
       "  /* The background is the default theme color */\n",
       "  color: var(--sklearn-color-text-on-default-background);\n",
       "}\n",
       "\n",
       "/* On hover, darken the color of the background */\n",
       "#sk-container-id-2 div.sk-label:hover label.sk-toggleable__label {\n",
       "  color: var(--sklearn-color-text);\n",
       "  background-color: var(--sklearn-color-unfitted-level-2);\n",
       "}\n",
       "\n",
       "/* Label box, darken color on hover, fitted */\n",
       "#sk-container-id-2 div.sk-label.fitted:hover label.sk-toggleable__label.fitted {\n",
       "  color: var(--sklearn-color-text);\n",
       "  background-color: var(--sklearn-color-fitted-level-2);\n",
       "}\n",
       "\n",
       "/* Estimator label */\n",
       "\n",
       "#sk-container-id-2 div.sk-label label {\n",
       "  font-family: monospace;\n",
       "  font-weight: bold;\n",
       "  display: inline-block;\n",
       "  line-height: 1.2em;\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-label-container {\n",
       "  text-align: center;\n",
       "}\n",
       "\n",
       "/* Estimator-specific */\n",
       "#sk-container-id-2 div.sk-estimator {\n",
       "  font-family: monospace;\n",
       "  border: 1px dotted var(--sklearn-color-border-box);\n",
       "  border-radius: 0.25em;\n",
       "  box-sizing: border-box;\n",
       "  margin-bottom: 0.5em;\n",
       "  /* unfitted */\n",
       "  background-color: var(--sklearn-color-unfitted-level-0);\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-estimator.fitted {\n",
       "  /* fitted */\n",
       "  background-color: var(--sklearn-color-fitted-level-0);\n",
       "}\n",
       "\n",
       "/* on hover */\n",
       "#sk-container-id-2 div.sk-estimator:hover {\n",
       "  /* unfitted */\n",
       "  background-color: var(--sklearn-color-unfitted-level-2);\n",
       "}\n",
       "\n",
       "#sk-container-id-2 div.sk-estimator.fitted:hover {\n",
       "  /* fitted */\n",
       "  background-color: var(--sklearn-color-fitted-level-2);\n",
       "}\n",
       "\n",
       "/* Specification for estimator info (e.g. \"i\" and \"?\") */\n",
       "\n",
       "/* Common style for \"i\" and \"?\" */\n",
       "\n",
       ".sk-estimator-doc-link,\n",
       "a:link.sk-estimator-doc-link,\n",
       "a:visited.sk-estimator-doc-link {\n",
       "  float: right;\n",
       "  font-size: smaller;\n",
       "  line-height: 1em;\n",
       "  font-family: monospace;\n",
       "  background-color: var(--sklearn-color-background);\n",
       "  border-radius: 1em;\n",
       "  height: 1em;\n",
       "  width: 1em;\n",
       "  text-decoration: none !important;\n",
       "  margin-left: 1ex;\n",
       "  /* unfitted */\n",
       "  border: var(--sklearn-color-unfitted-level-1) 1pt solid;\n",
       "  color: var(--sklearn-color-unfitted-level-1);\n",
       "}\n",
       "\n",
       ".sk-estimator-doc-link.fitted,\n",
       "a:link.sk-estimator-doc-link.fitted,\n",
       "a:visited.sk-estimator-doc-link.fitted {\n",
       "  /* fitted */\n",
       "  border: var(--sklearn-color-fitted-level-1) 1pt solid;\n",
       "  color: var(--sklearn-color-fitted-level-1);\n",
       "}\n",
       "\n",
       "/* On hover */\n",
       "div.sk-estimator:hover .sk-estimator-doc-link:hover,\n",
       ".sk-estimator-doc-link:hover,\n",
       "div.sk-label-container:hover .sk-estimator-doc-link:hover,\n",
       ".sk-estimator-doc-link:hover {\n",
       "  /* unfitted */\n",
       "  background-color: var(--sklearn-color-unfitted-level-3);\n",
       "  color: var(--sklearn-color-background);\n",
       "  text-decoration: none;\n",
       "}\n",
       "\n",
       "div.sk-estimator.fitted:hover .sk-estimator-doc-link.fitted:hover,\n",
       ".sk-estimator-doc-link.fitted:hover,\n",
       "div.sk-label-container:hover .sk-estimator-doc-link.fitted:hover,\n",
       ".sk-estimator-doc-link.fitted:hover {\n",
       "  /* fitted */\n",
       "  background-color: var(--sklearn-color-fitted-level-3);\n",
       "  color: var(--sklearn-color-background);\n",
       "  text-decoration: none;\n",
       "}\n",
       "\n",
       "/* Span, style for the box shown on hovering the info icon */\n",
       ".sk-estimator-doc-link span {\n",
       "  display: none;\n",
       "  z-index: 9999;\n",
       "  position: relative;\n",
       "  font-weight: normal;\n",
       "  right: .2ex;\n",
       "  padding: .5ex;\n",
       "  margin: .5ex;\n",
       "  width: min-content;\n",
       "  min-width: 20ex;\n",
       "  max-width: 50ex;\n",
       "  color: var(--sklearn-color-text);\n",
       "  box-shadow: 2pt 2pt 4pt #999;\n",
       "  /* unfitted */\n",
       "  background: var(--sklearn-color-unfitted-level-0);\n",
       "  border: .5pt solid var(--sklearn-color-unfitted-level-3);\n",
       "}\n",
       "\n",
       ".sk-estimator-doc-link.fitted span {\n",
       "  /* fitted */\n",
       "  background: var(--sklearn-color-fitted-level-0);\n",
       "  border: var(--sklearn-color-fitted-level-3);\n",
       "}\n",
       "\n",
       ".sk-estimator-doc-link:hover span {\n",
       "  display: block;\n",
       "}\n",
       "\n",
       "/* \"?\"-specific style due to the `<a>` HTML tag */\n",
       "\n",
       "#sk-container-id-2 a.estimator_doc_link {\n",
       "  float: right;\n",
       "  font-size: 1rem;\n",
       "  line-height: 1em;\n",
       "  font-family: monospace;\n",
       "  background-color: var(--sklearn-color-background);\n",
       "  border-radius: 1rem;\n",
       "  height: 1rem;\n",
       "  width: 1rem;\n",
       "  text-decoration: none;\n",
       "  /* unfitted */\n",
       "  color: var(--sklearn-color-unfitted-level-1);\n",
       "  border: var(--sklearn-color-unfitted-level-1) 1pt solid;\n",
       "}\n",
       "\n",
       "#sk-container-id-2 a.estimator_doc_link.fitted {\n",
       "  /* fitted */\n",
       "  border: var(--sklearn-color-fitted-level-1) 1pt solid;\n",
       "  color: var(--sklearn-color-fitted-level-1);\n",
       "}\n",
       "\n",
       "/* On hover */\n",
       "#sk-container-id-2 a.estimator_doc_link:hover {\n",
       "  /* unfitted */\n",
       "  background-color: var(--sklearn-color-unfitted-level-3);\n",
       "  color: var(--sklearn-color-background);\n",
       "  text-decoration: none;\n",
       "}\n",
       "\n",
       "#sk-container-id-2 a.estimator_doc_link.fitted:hover {\n",
       "  /* fitted */\n",
       "  background-color: var(--sklearn-color-fitted-level-3);\n",
       "}\n",
       "</style><div id=\"sk-container-id-2\" class=\"sk-top-container\"><div class=\"sk-text-repr-fallback\"><pre>RandomForestClassifier(random_state=42)</pre><b>In a Jupyter environment, please rerun this cell to show the HTML representation or trust the notebook. <br />On GitHub, the HTML representation is unable to render, please try loading this page with nbviewer.org.</b></div><div class=\"sk-container\" hidden><div class=\"sk-item\"><div class=\"sk-estimator fitted sk-toggleable\"><input class=\"sk-toggleable__control sk-hidden--visually\" id=\"sk-estimator-id-2\" type=\"checkbox\" checked><label for=\"sk-estimator-id-2\" class=\"sk-toggleable__label fitted sk-toggleable__label-arrow fitted\">&nbsp;&nbsp;RandomForestClassifier<a class=\"sk-estimator-doc-link fitted\" rel=\"noreferrer\" target=\"_blank\" href=\"https://scikit-learn.org/1.5/modules/generated/sklearn.ensemble.RandomForestClassifier.html\">?<span>Documentation for RandomForestClassifier</span></a><span class=\"sk-estimator-doc-link fitted\">i<span>Fitted</span></span></label><div class=\"sk-toggleable__content fitted\"><pre>RandomForestClassifier(random_state=42)</pre></div> </div></div></div></div>"
      ],
      "text/plain": [
       "RandomForestClassifier(random_state=42)"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clf = RandomForestClassifier(random_state=42)\n",
    "clf.fit(X_train, y_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "632da487-9d71-4f5a-8b47-d2cc8b321c8b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy: 1.0\n",
      "Classification Report:\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "           0       1.00      1.00      1.00       345\n",
      "           1       1.00      1.00      1.00        12\n",
      "\n",
      "    accuracy                           1.00       357\n",
      "   macro avg       1.00      1.00      1.00       357\n",
      "weighted avg       1.00      1.00      1.00       357\n",
      "\n"
     ]
    }
   ],
   "source": [
    "y_pred = clf.predict(X_test)\n",
    "\n",
    "print(\"Accuracy:\", accuracy_score(y_test, y_pred))\n",
    "print(\"Classification Report:\\n\", classification_report(y_test, y_pred))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "b3711115-2c03-4c5d-a17b-da2d805bb14a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "channel           sample   A[C>A]A   A[C>A]C   A[C>A]G  A[C>A]T   A[C>G]A  \\\n",
      "0        TCGA-A2-A0T5-01  0.000000  0.000000  0.000000  0.00000  0.000000   \n",
      "1        TCGA-CF-A9FF-01  0.000000  0.010811  0.010811  0.00000  0.000000   \n",
      "2        TCGA-CA-6717-01  0.000848  0.001774  0.000309  0.01049  0.000848   \n",
      "\n",
      "channel   A[C>G]C   A[C>G]G  A[C>G]T   A[C>T]A  ...   T[T>A]T   T[T>C]A  \\\n",
      "0        0.000000  0.000000  0.00000  0.000000  ...  0.000000  0.000000   \n",
      "1        0.000000  0.010811  0.00000  0.010811  ...  0.000000  0.005405   \n",
      "2        0.000848  0.000000  0.00054  0.001465  ...  0.002854  0.006788   \n",
      "\n",
      "channel   T[T>C]C   T[T>C]G   T[T>C]T   T[T>G]A   T[T>G]C  T[T>G]G   T[T>G]T  \\\n",
      "0        0.000000  0.000000  0.000000  0.000000  0.000000   0.0000  0.000000   \n",
      "1        0.000000  0.000000  0.000000  0.000000  0.000000   0.0000  0.005405   \n",
      "2        0.017123  0.009873  0.017663  0.007636  0.009873   0.0027  0.082298   \n",
      "\n",
      "channel  p53 status  \n",
      "0                    \n",
      "1                    \n",
      "2                    \n",
      "\n",
      "[3 rows x 98 columns]\n",
      "\n",
      "--------------------------\n",
      "\n",
      "                HG19_Variant  channel        Pathogenicity  p53 status\n",
      "128974    chr17:g.7577546C>T  G[C>T]C                  VUS        0.25\n",
      "142812    chr17:g.7577555C>A  T[C>A]C           Pathogenic        1.00\n",
      "192463    chr17:g.7577093G>A  C[G>A]G  Possibly pathogenic        0.50\n",
      "260865    chr17:g.7578265T>A  A[T>A]C    Likely Pathogenic        0.75\n",
      "441994    chr17:g.7577093G>A  C[G>A]G  Possibly pathogenic        0.50\n",
      "...                      ...      ...                  ...         ...\n",
      "20327269  chr17:g.7578393T>C  C[T>C]T               Benign        0.00\n",
      "20345243  chr17:g.7574017G>A  C[G>A]C  Possibly pathogenic        0.50\n",
      "20389252  chr17:g.7578405C>T  C[C>T]C               Benign        0.00\n",
      "20399949  chr17:g.7578211G>A  C[G>A]A           Pathogenic        1.00\n",
      "20491625  chr17:g.7578507C>T  T[C>T]C                  VUS        0.25\n",
      "\n",
      "[176 rows x 4 columns]\n",
      "\n",
      "--------------------------\n",
      "\n",
      "[[-0.88615687 -0.73758267 -0.70710678 -0.70710678 -0.70710678 -0.70710678\n",
      "  -0.70710678 -0.85459103 -1.41421285  1.40948096 -0.70710678  1.41417916\n",
      "  -0.70710678 -0.70710678 -0.70710678 -0.70710678 -0.70710678 -1.29383567\n",
      "  -1.2522773  -0.70710678 -0.70710678 -0.70710678 -0.91700014 -1.23576888\n",
      "  -1.07841634 -0.8624794  -0.70710678  1.41421143  0.         -0.70710678\n",
      "  -0.70710678 -0.70710678 -0.70710678  1.2974344   1.41146054 -0.70710678\n",
      "  -0.70710678 -0.70710678 -0.70710678 -0.70710678 -0.70710678 -0.70710678\n",
      "   1.41013098 -0.70710678 -0.70710678  1.40846286 -0.89604924  1.41415971\n",
      "  -0.70710678 -0.70710678 -0.70710678 -0.70710678 -0.84670725  0.\n",
      "  -0.70710678 -0.70710678 -1.41408541  1.38531514 -1.31362764  0.\n",
      "  -0.81524702 -0.70710678 -0.70710678 -0.70710678 -1.34320737 -0.70710678\n",
      "  -1.01816158 -0.70710678 -0.70710678 -0.70710678 -1.40954253 -1.41333154\n",
      "  -1.23952569 -1.08568803 -0.76625413 -0.70710678 -0.71143538 -0.70710678\n",
      "  -0.71089383 -0.2086989  -0.91474362 -0.851538   -1.11046667 -0.70710678\n",
      "  -0.70710678 -0.70710678 -0.70710678 -1.38772056 -0.70710678 -0.70710678\n",
      "  -0.70710678 -0.70710678 -0.70710678 -0.70710678 -0.77779296  1.\n",
      "   0.          0.        ]\n",
      " [ 1.39756495  1.41376926 -0.70710678 -0.70710678 -0.70710678  1.41421356\n",
      "  -0.70710678  1.40313138  0.70833428 -0.60462772 -0.70710678 -0.71563182\n",
      "  -0.70710678 -0.70710678 -0.70710678 -0.70710678 -0.70710678  0.15245657\n",
      "   0.05705941 -0.70710678 -0.70710678 -0.70710678 -0.47388025  1.21341484\n",
      "   1.33152354  1.4018565  -0.70710678 -0.70923489  0.          1.41421356\n",
      "  -0.70710678 -0.70710678 -0.70710678 -0.16137954 -0.6293473  -0.70710678\n",
      "  -0.70710678 -0.70710678 -0.70710678 -0.70710678 -0.70710678 -0.70710678\n",
      "  -0.79805988 -0.70710678 -0.70710678 -0.81456855 -0.49950998 -0.71776827\n",
      "  -0.70710678 -0.70710678 -0.70710678 -0.70710678  1.40433024  0.\n",
      "  -0.70710678 -0.70710678  0.72353029 -0.9389835   0.20317649  0.\n",
      "   1.40838783 -0.70710678 -0.70710678 -0.70710678  0.28839946 -0.70710678\n",
      "  -0.34092524 -0.70710678 -0.70710678 -0.70710678  0.60531071  0.66341686\n",
      "   1.20940854  1.32767593 -0.64626156  1.41421356  1.41420471  1.41421356\n",
      "   1.41420679  1.3156849   1.39141307 -0.55206651  1.31361835 -0.70710678\n",
      "  -0.70710678 -0.70710678 -0.70710678  0.45790745 -0.70710678 -0.70710678\n",
      "  -0.70710678 -0.70710678 -0.70710678 -0.70710678 -0.63398112  0.\n",
      "   0.          1.        ]\n",
      " [-0.51140807 -0.67618659  1.41421356  1.41421356  1.41421356 -0.70710678\n",
      "   1.41421356 -0.54854035  0.70587857 -0.80485324  1.41421356 -0.69854734\n",
      "   1.41421356  1.41421356  1.41421356  1.41421356  1.41421356  1.1413791\n",
      "   1.19521789  1.41421356  1.41421356  1.41421356  1.39088039  0.02235404\n",
      "  -0.2531072  -0.53937709  1.41421356 -0.70497653  0.         -0.70710678\n",
      "   1.41421356  1.41421356  1.41421356 -1.13605485 -0.78211324  1.41421356\n",
      "   1.41421356  1.41421356  1.41421356  1.41421356  1.41421356  1.41421356\n",
      "  -0.61207111  1.41421356  1.41421356 -0.59389431  1.39555922 -0.69639144\n",
      "   1.41421356  1.41421356  1.41421356  1.41421356 -0.55762299  0.\n",
      "   1.41421356  1.41421356  0.69055512 -0.44633165  1.11045115  0.\n",
      "  -0.5931408   1.41421356  1.41421356  1.41421356  1.05480791  1.41421356\n",
      "   1.35908682  1.41421356  1.41421356  1.41421356  0.80423182  0.74991468\n",
      "   0.03011716 -0.2419879   1.4125157  -0.70710678 -0.70276933 -0.70710678\n",
      "  -0.70331296 -1.106986   -0.47666946  1.40360451 -0.20315169  1.41421356\n",
      "   1.41421356  1.41421356  1.41421356  0.92981311  1.41421356  1.41421356\n",
      "   1.41421356  1.41421356  1.41421356  1.41421356  1.41177408  0.\n",
      "   1.          0.        ]]\n",
      "\n",
      "--------------------------\n",
      "\n",
      "<Compressed Sparse Row sparse matrix of dtype 'float64'\n",
      "\twith 704 stored elements and shape (176, 81)>\n",
      "  Coords\tValues\n",
      "  (0, 0)\t-0.548894921121196\n",
      "  (0, 18)\t1.0\n",
      "  (0, 41)\t1.0\n",
      "  (0, 80)\t1.0\n",
      "  (1, 0)\t1.3330305227229045\n",
      "  (1, 23)\t1.0\n",
      "  (1, 42)\t1.0\n",
      "  (1, 78)\t1.0\n",
      "  (2, 0)\t0.07841356016017086\n",
      "  (2, 11)\t1.0\n",
      "  (2, 31)\t1.0\n",
      "  (2, 79)\t1.0\n",
      "  (3, 0)\t0.7057220414415377\n",
      "  (3, 2)\t1.0\n",
      "  (3, 58)\t1.0\n",
      "  (3, 77)\t1.0\n",
      "  (4, 0)\t0.07841356016017086\n",
      "  (4, 11)\t1.0\n",
      "  (4, 31)\t1.0\n",
      "  (4, 79)\t1.0\n",
      "  (5, 0)\t-1.176203402402563\n",
      "  (5, 15)\t1.0\n",
      "  (5, 59)\t1.0\n",
      "  (5, 76)\t1.0\n",
      "  (6, 0)\t-0.548894921121196\n",
      "  :\t:\n",
      "  (169, 77)\t1.0\n",
      "  (170, 0)\t1.3330305227229045\n",
      "  (170, 19)\t1.0\n",
      "  (170, 47)\t1.0\n",
      "  (170, 78)\t1.0\n",
      "  (171, 0)\t-1.176203402402563\n",
      "  (171, 15)\t1.0\n",
      "  (171, 63)\t1.0\n",
      "  (171, 76)\t1.0\n",
      "  (172, 0)\t0.07841356016017086\n",
      "  (172, 10)\t1.0\n",
      "  (172, 29)\t1.0\n",
      "  (172, 79)\t1.0\n",
      "  (173, 0)\t-1.176203402402563\n",
      "  (173, 8)\t1.0\n",
      "  (173, 66)\t1.0\n",
      "  (173, 76)\t1.0\n",
      "  (174, 0)\t1.3330305227229045\n",
      "  (174, 9)\t1.0\n",
      "  (174, 53)\t1.0\n",
      "  (174, 78)\t1.0\n",
      "  (175, 0)\t-0.548894921121196\n",
      "  (175, 25)\t1.0\n",
      "  (175, 72)\t1.0\n",
      "  (175, 80)\t1.0\n"
     ]
    }
   ],
   "source": [
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.preprocessing import StandardScaler, OneHotEncoder\n",
    "from sklearn.compose import ColumnTransformer\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "from sklearn.pipeline import Pipeline\n",
    "from sklearn.metrics import accuracy_score \n",
    "\n",
    "print(sig_data)\n",
    "\n",
    "X_sig = sig_data.drop(columns=['p53 status'])  # Drop target column in sig_data\n",
    "y = sig_data['p53 status']                     # Target label (p53 status)\n",
    "\n",
    "X_channel = prepared_data.drop(columns=['gene', 'Final comment'])\n",
    "print(\"\\n--------------------------\\n\")\n",
    "print(X_channel)\n",
    "\n",
    "numerical_features = sig_data.columns[2:-1]\n",
    "sig_preprocessor = ColumnTransformer([\n",
    "    ('scaler', StandardScaler(), numerical_features),\n",
    "    ('encoder', OneHotEncoder(), ['sample'])\n",
    "])\n",
    "\n",
    "channel_preprocessor = ColumnTransformer([\n",
    "    ('scaler', StandardScaler(), ['p53 status']),\n",
    "    ('encoder', OneHotEncoder(), ['channel', 'HG19_Variant', 'Pathogenicity'])\n",
    "])\n",
    "\n",
    "X_sig_transformed = sig_preprocessor.fit_transform(X_sig)\n",
    "X_channel_transformed = channel_preprocessor.fit_transform(X_channel)\n",
    "\n",
    "print(\"\\n--------------------------\\n\")\n",
    "print(X_sig_transformed)\n",
    "print(\"\\n--------------------------\\n\")\n",
    "print(X_channel_transformed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "5e1609e9-e454-449b-82cf-b6ebffd8039e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # prepare input data\n",
    "# from sklearn.preprocessing import OneHotEncoder\n",
    "\n",
    "# # prepare input data\n",
    "# def prepare_inputs(X_train, X_test):\n",
    "# \tohe = OneHotEncoder()\n",
    "# \tohe.fit(X_train)\n",
    "# \tX_train_enc = ohe.transform(X_train)\n",
    "# \tX_test_enc = ohe.transform(X_test)\n",
    "# \treturn X_train_enc, X_test_enc\n",
    "\n",
    "# # prepare target\n",
    "# def prepare_targets(y_train, y_test):\n",
    "# \tle = LabelEncoder()\n",
    "# \tle.fit(y_train)\n",
    "# \ty_train_enc = le.transform(y_train)\n",
    "# \ty_test_enc = le.transform(y_test)\n",
    "# \treturn y_train_enc, y_test_enc\n",
    "\n",
    "# # encoding not working?\n",
    "\n",
    "\n",
    "# # prepare input data\n",
    "# X_train_enc, X_test_enc = prepare_inputs(X_train, X_test)\n",
    "# # prepare output data\n",
    "# # y_train_enc, y_test_enc = prepare_targets(y_train, y_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1e4c6834-e459-4d19-87ba-64233cf0238c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
